src/main/java/edu/umich/ctools/sectionsUtilityTool/Friend.java

```diff
@@ -0,0 +1,317 @@
+			}
+		}
+		catch (Exception e) {
+			M_log.warn("Friend checkAccountExist(): email address " + email + " " + e.getMessage());
+			e.printStackTrace();
+		}
+		return rv;
+	}
+
+	public static String readFile(String path, Charset encoding) throws IOException{
+		byte[] encoded = Files.readAllBytes(Paths.get(path));
+		return new String(encoded, encoding);
+	}
+	
+	
+	/*
+	 * Actually sends out an invite
+	 *
+	 * @param accountEmail friend account to be invited
+	 * @param currentUser the person doing the inviting
+	 * 
+	 * @param associated site id
+	 * @return
+	 * -1 = bad e-mail address
+	 * 0 = runtime problem
+	 * 1 = successfully sent invitation 
+	 */
+
+	//accountEmail == userToBeInvited
+	//currentUserEmail == instructorEmail
+	public static int doSendInvite(String accountEmail, 
+			String currentUserEmail, 
+			String instructorName) {
+		int rv = 0; // default to 
+		
+		try {
+			emailMessage = Friend.readFile(friendEmailFile, StandardCharsets.UTF_8);
+			emailMessage = emailMessage.replace("<instructor>", instructorName);
+			emailMessage = emailMessage.replace("<contactEmail>", contactEmail);
+			
+			Xclient = new XmlRpcClient();
+			XmlRpcClientConfigImpl config = new XmlRpcClientConfigImpl();
+			config.setServerURL(new URL(friendUrl));
+			Xclient.setTransportFactory(new XmlRpcSun15HttpTransportFactory(Xclient));
+			Xclient.setConfig(config);
+			
+			Object[] params = new Object[]{contactEmail, referrerUrl, emailMessage, new String[]{accountEmail}, currentUserEmail};
+			Object[] results = (Object[]) Xclient.execute("sendInvites", params);
+			// though the friend XML-RPC service supports batch call mode, 
+			// due to the ctools event model ( one "user added event" per user), 
+			// we will pass only one email address to the "sendInvites" call for now
+			// hence we should only get one call result returned
+			if (results != null && results.length == 1)
+			{
+				rv = ((Integer) results[0]).intValue();
+			}
+		}
+		catch (Exception e) {
+			M_log.warn("Friend doSendInvite(): email address=" + accountEmail + " " + e.getMessage());
+		}
+
+		return rv;
+	}    
+
+	public static void notifyCurrentUser(String instructorName, String instructorEmail, String inviteEmail){
+
+		String to = instructorEmail;
+		String from = contactEmail;
+		String subjectLine = "Friend Accounts Integration Email";
+		//This will need to be changed to properties
+		String host = "deluxe.dsc.umich.edu";
+
+		System.out.println("Setting up props");
+		Properties properties = System.getProperties();
+		properties.put("mail.smtp.auth", "false");
+		properties.put("mail.smtp.starttls.enable", "true"); //Put below to false, if no https is needed
+		properties.put("mail.smtp.host", host);
+		properties.put("mail.debug", "true");
+
+		System.out.println("Initiating Session");
+		Session session = Session.getInstance(properties);
+
+		try{
+			
+			emailMessage = Friend.readFile(requesterEmailFile, StandardCharsets.UTF_8);
```

Comment:
Could have a single method that would read a file and replace all the macros that it finds.